WordPress/src/main/java/org/wordpress/android/ui/media/MediaBrowserActivity.java

```diff
@@ -501,14 +505,36 @@ public void onBackPressed() {
         }
     }
 
-    @Override
-    public void onMediaAdded(String mediaId) {
-        if (WordPress.getCurrentBlog() == null || mediaId == null) {
+    @SuppressWarnings("unused")
+    public void onEventMainThread(MediaEvents.MediaStateChanged event) {
+        updateOnMediaChanged(event.mLocalBlogId, event.mMediaId);
+    }
+
+    @SuppressWarnings("unused")
+    public void onEventMainThread(MediaEvents.MediaAdded event) {
+        updateOnMediaChanged(event.mLocalBlogId, event.mMediaId);
+    }
+
+    @SuppressWarnings("unused")
+    public void onEventMainThread(MediaEvents.MediaUploadSucceed event) {
+        updateOnMediaChanged(event.mLocalBlogId, event.mLocalMediaId);
+    }
+
+    @SuppressWarnings("unused")
+    public void onEventMainThread(MediaEvents.MediaFetched event) {
+        updateOnMediaChanged(event.mLocalBlogId, event.mMediaId);
+    }
+
+    @SuppressWarnings("unused")
+    public void onEventMainThread(MediaEvents.MediaUploadFailed event) {
+        ToastUtils.showToast(this, event.mErrorMessage, ToastUtils.Duration.SHORT);
+    }
+
+    public void updateOnMediaChanged(String blogId, String mediaId) {
+        if (mediaId == null) {
             return;
         }
-        String blogId = String.valueOf(WordPress.getCurrentBlog().getLocalTableBlogId());
         Cursor cursor = WordPress.wpDB.getMediaFile(blogId, mediaId);
```

Comment:
The addition of the event(s) causes this routine to be called quite frequently, and it's opening a cursor simply to check whether the media file exists. Perhaps it'd be more efficient to add a routine like this to `WordPressDB`?

```
public boolean mediaFileExists(String blogId, String mediaId) {
    return SqlUtils.boolForQuery(db, 
            "SELECT 1 FROM " + MEDIA_TABLE + " WHERE blogId=? AND mediaId=?", 
            new String[]{blogId, mediaId});
}
```